packages/front-end/components/Features/FeatureRules.tsx

```diff
@@ -81,79 +82,98 @@ export default function FeatureRules({
     })
   );
 
+  const [tabEnvs, dropdownEnvs] = useMemo(() => {
+    const tabEnvs: Environment[] = [];
+    const dropdownEnvs: Environment[] = [];
+    environments.forEach((env) => {
+      if (env.toggleOnList) {
```

Comment:
Thinking about this more, it might be better to just always render the first X environments instead of using `toggleOnList`.  There are valid reasons for having this setting off for some important environments.  For example, if you don't want people to accidentally toggle something on production.

It's also good to add some special case logic so we don't end up with a dropdown with a single item in it.  Something like

```ts
if (environments.length <= 5 || i < 5) {
  // tab
} else {
  // dropdown
}
```